tests/phantoms/_SaturationRecoveryEllipsePhantom.py

```diff
@@ -0,0 +1,86 @@
+"""Saturation recovery ellipse phantom for testing."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import torch
+
+from mrpro.data import SpatialDimension
+from mrpro.operators import Operator
+from mrpro.operators.models import SaturationRecovery
+from mrpro.phantoms import EllipsePhantom
+from tests._RandomGenerator import RandomGenerator
+
+
+# TODO: maybe make it possible to use more and different ellipses?
+class SaturationRecoveryEllipsePhantom2D:
```

Comment:
do we need to be this fancy here?

Do we need a special saturation recovery phantom?

Reply:
if we want to overly a polynial and phase to an ellipse phantom, maybe we can do this in a more general form instead the phantom?

Reply:
i removed the entire test as it does not test the functionality of the functions but it could rather be used as an example. I replace it with a one where `lbfgs `and `adam `are tested also using the `ConstraintsOp`